### DISCUSSION TITLE
More information about subscripts and projections

### DISCUSSION THREAD

**Discussion by mspiegel** [hylo-lang/hylo#1234] [Q&A] [answered]
Hello,

What resources would you recommend if I am interested in gaining a better understanding of subscripts and projections? The language tour has an introduction to subscripts. The language specification provides some information about them. I understand that ‘yield’ in a subscript will produce a projection. Are there other parts of the language that produce projections? How are projections implemented? Silly question but can subscripts be recursive? I’m eager to learn more.

**Comment by dabrahams** [comment, marked as answer]
Yes, they can be recursive.  There's a simple rewrite rule explained [here](https://youtu.be/GsxYnEAZoNI?si=6BoJrfh--E4XpWPb&t=259) that shows how projections are implemented in terms of callbacks. If you still have questions after watching that, please ask!

**Reply by mspiegel** [reply]
Thank you!

**Comment by tothambrus11** [comment]
For the record: We have a new mental model and lowering scheme to think about subscripts, described in detail in this document: https://hackmd.io/@lucteo/ByYUkw3QWl, and in this Swift thread: https://forums.swift.org/t/efficient-yielding-accessor-abi/82891/11
